Pinta.Core/Classes/Document.cs

```diff
@@ -67,9 +79,34 @@ public Document (Size size)
 		HasBeenSavedInSession = false;
 		ImageSize = size;
 
+		// --- Post-initialization 
+
+		if (file is not null) {
+
+			if (string.IsNullOrEmpty (fileType))
+				throw new ArgumentNullException ($"nameof{fileType} must contain value.");
+
+			File = file;
+			FileType = fileType;
+		} else
+			DisplayName = Translations.GetString ("Unsaved Image {0}", name_counter++);
+
 		ResetSelectionPaths ();
 	}
 
+	// This pair of methods is temporary, in order to ensure that a document is attached to a single workspace at a time
+	// While it exists, it is to be called exclusively from the workspace manager
+	private bool is_attached = false; // By default detached from workspace
```

Comment:
This check doesn't really seem necessary to keep around IMO, since there's only one workspace?

Reply:
This is intended to catch certain types of bugs. Regardless, I removed it in the last commit